### DISCUSSION TITLE
How to get the oldest timestamp? / How old is my oldest metric?

### DISCUSSION THREAD

**Discussion by applejag** [grafana/mimir#5487] [Help and support] [answered]
Prometheus exports the metric `prometheus_tsdb_lowest_timestamp_seconds` which is the timestamp of the oldest metric.

I cannot find an equivalent metric on Mimir. Is there an equivalent metric, or can I get this information a different way?

**Comment by pstibrany** [comment, marked as answer]
There is no equivalent, and no easy way to get this information. Mimir would need to compute this from blocks and TSDB Head in each ingester. There is no component doing that right now.

With access to the block storage, one could find the oldest block and take `minT` of that block. This is easily available in `bucket-index.json` file. This approach would ignore data from ingesters, but assuming that data on the storage is older, that may not be a problem.
